datasets/flwr_datasets/partitioner/distribution_partitioner.py

```diff
@@ -0,0 +1,423 @@
+        self._partition_id_to_indices_determined = True
+
+    def _check_distribution_array_shape_if_needed(self) -> None:
+        """Test distribution array shape correctness."""
+        if not self._partition_id_to_indices_determined:
+            if not isinstance(self._distribution_array, np.ndarray):
+                raise TypeError("Input must be a numpy array.")
+
+            if self._distribution_array.ndim != 2:
+                raise ValueError("The distribution array is not 2-dimensional.")
+
+            self._num_unique_labels = len(self.dataset.unique(self._partition_by))
+            self._num_columns = int(
+                self._num_unique_labels_per_partition
+                * self._num_partitions
+                / self._num_unique_labels
+            )
+
+            if self._distribution_array.shape[0] != self._num_unique_labels:
+                raise ValueError(
+                    "The expected number of rows in `distribution_array` must equal to "
+                    "the number of unique labels in the dataset, which is "
+                    f"{self._num_unique_labels}, but the number of rows in "
+                    f"`distribution_array` is {self._distribution_array.shape[0]}."
+                )
+
+            if self._distribution_array.shape[1] != self._num_columns:
+                raise ValueError(
+                    "The expected number of columns in `distribution_array` is "
+                    f"{self._num_columns} (refer to the documentation for the "
+                    "expression), but the number of columns in `distribution_array` "
+                    f"is {self._distribution_array.shape[0]}."
+                )
+
+    def _check_num_unique_labels_per_partition_if_needed(self) -> None:
+        """Test number of unique labels do not exceed self.num_unique_labels."""
+        if self._num_unique_labels_per_partition > self._num_unique_labels:
+            raise ValueError(
+                "The specified `num_unique_labels_per_partition`"
+                f"={self._num_unique_labels_per_partition} is greater than the number "
+                f"of unique classes in the given dataset={self._num_unique_labels}. "
+                "Reduce the `num_unique_labels_per_partition` or make use different "
+                "dataset to apply this partitioning."
+            )
+
+    def _check_distribution_array_sum_if_needed(self) -> None:
+        """Test correctness of distribution array sum."""
+        if not self._partition_id_to_indices_determined and not self._rescale:
+            labels = self.dataset[self._partition_by]
+            distribution = sorted(Counter(labels).items())
+            distribution_vals = [v for _, v in distribution]
+
+            if any(self._distribution_array.sum(1) > distribution_vals):
+                raise ValueError(
+                    "The sum of at least one label distribution array "
+                    "exceeds the original class label distribution."
```

Comment:
```suggestion
                    "exceeds that of the original class label distribution."
```

Reply:
i'm not sure if the part of "original class label distribution" is accurate. See previous comment in this method.

Reply:
I've reworded it to "... exceeds that of the unique labels counter in the given dataset ..."